include/curl/curl.h

```diff
@@ -133,6 +133,15 @@ typedef void CURLSH;
 #  define CURL_EXTERN
 #endif
 
+#if !defined(CURL_FALLTHROUGH)
+#if (defined(__GNUC__) && __GNUC__ >= 7) || \
+    (defined(__clang__) && __clang_major__ >= 10)
+#  define CURL_FALLTHROUGH()  __attribute__((fallthrough))
+#else
+#  define CURL_FALLTHROUGH()  do {} while (0)
+#endif
+#endif
```

Comment:
Ah right, is just there to make this macro use work...

Reply:
Good point about the placement, I didn't gave a thought at all (and that while copying this from libssh2's internal header). Definitely needs to be local, I'll move it.

Reply:
examples are also compiled in CI: https://github.com/curl/curl/actions/runs/7146212736/job/19463469167?pr=12489#step:32:36

Reply:
Yes, but I don't think that is reason enough for us to add this to the public headers. We should probably rather try to
1. avoid the fallthroughs in example code or
2. avoid the warning in example code

Reply:
I could add that I previously have avoided this particular warning just because I think this "clutter" is bad for readability.

Reply:
The clutter was already there as `/* FALLTHROUGH */` comments, the problem was that newer compiler versions no longer honored it, so we had to disable the warning for them. This has been solved since on the compiler end.

Reply:
Looking into the code, it was easy to avoid those fallthroughs.

Reply:
I think /* FALLTHROUGH */ comments was a rather elegant way. Oh well, adapt and overcome.

Reply:
Doc [lists](https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wimplicit-fallthrough_003d) 11 lines of regexps to define the accepted legacy syntax, spread out to 4 different modes. And this is just gcc. Just grepping them was non-trivial! I'm happy gcc and llvm could agree on a single, common syntax.